source/blog/_posts/2017-09-25-ajax-panel.md

```diff
@@ -0,0 +1,130 @@
+---
+title: Ajax Panel
+tags:
+- frontend
+- ajax
+- javascript
+- jquery
+- framework
+
+categories:
+- dev
+
+authors: [lh]
+github_link: blog/_posts/2017-09-25-ajax-panel.md
+
+---
+
+In this blog post I want to present the concept of our ajax based panel system which we use in our B2B-Suite.
+In the B2B-Suite we had to develop a backend user interface inside the shopware frontend. 
+
+On a single page in a typical backend view you may find:
+* list of entities
+* form to enter new data
+* list of related entities
+* some statistics
+
+Serving all this data from a single controller action may already be hard, now imagine the sheer number of parameters you may 
+have to exchange with this single action when you want to enable pagination, validation and search on this single page, 
+through this single action.
+
+This is pretty much the reason why stateless services and statefull frontends are a important topic in todays web development. 
+And it is exactly the reason why we created the ajax panel. It provides us with the means to load local state from the 
+server and create a rich ui.
+
+We evaluated different frameworks to achieve this target. AngularJS and Vue.JS were possible frameworks which allows
+two way data binding and statefull frontend access.
+
+<img src="/blog/img/ajax-panel-abstract.svg" alt="image">
+
+The small controller actions doesn't respond with the full page dom tree anymore. Each controller is only responsible for a
+specific panel content.
+
+Our target was also to use most of the already existing dependencies instead of adding new frameworks just for
+the B2B-Suite. As you maybe already know jQuery is a base dependency of Shopware 5.
+
+We decided to develop our own lightweight frontend framework on top of jQuery which allows asynchronous HTTP requests for our frontend.
+The main target of this framework is to execute the asynchronous call and render the response in a given and zoned DOM element by using event.
```

Comment:
"to execute asynchronous calls". 

mit "the" hört sich das merkwürdig spezifisch an, so als gäbe es da einen konkreten Call, den du meinst